src/store/cockpit/cockpitApi.ts

```diff
@@ -503,14 +503,35 @@ export const cockpitApi = contentSourcesApi.injectEndpoints({
                 image_requests: [ir],
               };
 
+              // we need a customizations object rather than a blueprint
+              // since we need to set the `subscription` options for
+              // registering an image. This is not available with blueprint
+              // customizations
+              const customizations = mapHostedToOnPrem(
+                blueprint as CreateBlueprintRequest,
+              ).customizations as Customizations;
+
+              if (blueprint.customizations.subscription) {
```

Comment:
What if we change `mapHostedToOnPrem` to return a `ComposeRequest` or even just `Customizations` as defined in `src/store/cockpit/composerCloudApi.ts` ? Then we can move this block into that function as well.

 The only place `mapHostedToOnPrem` is used is here afaict, so it's really just a helper function.